src/docs/development/ui/navigation/deep-linking.md

```diff
@@ -0,0 +1,116 @@
+<array>
+    <dict>
+    <key>CFBundleTypeRole</key>
+    <string>Editor</string>
+    <key>CFBundleURLName</key>
+    <string>flutterbooksample.com</string>
+    <key>CFBundleURLSchemes</key>
+    <array>
+    <string>customscheme</string>
+    </array>
+    </dict>
+</array>
+```
+
+The CFBundleURLName is a unique URL used to distinguish your app from others
+that use the same scheme. The scheme (`customscheme://`)  can also be unique to
+help avoid conflicts.
+
+A full restart is required to apply these changes.
+
+## Test on iOS Simulator
+The `xcrun` command can be used to test on the iOS Simulator:
+
+```
+xcrun simctl openurl booted customscheme://flutterbooksample.com/book/1 
+```
+
+## Migrating from plugin-based deep linking
+
+If you have written a plugin to handle deep links, as described in [this
+article][plugin-linking], it will continue to work until you opt-in to this
+behavior by adding `FlutterDeepLinkingEnabled` to `Info.plist` or
+`flutter_deeplinking_enabled` to `AndroidManifest.xml`, respectively.
+
+## Behavior
+
+The behavior varies slightly based on the platform and whether or not the app is
+launched and running.
+
+| Platform / Scenario      | Using Navigator                                                     | Using Router                                                                                                                                                                                        |
+|--------------------------|---------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| iOS (not launched)       | App gets initialRoute ("/") and a short time after gets a pushRoute | App gets initialRoute ("/") and a short time after uses the RouteInformationParser to parse the route and call RouterDelegate.setNewRoutePath, which configures the Navigator with the corresponding Page. |
+| Android - (not launched) | App gets initialRoute containing the full route ("http:/deeplink/") | App gets initialRoute ("/deeplink") and passes it to the RouteInformationParser to parse the route and call RouterDelegate.setNewRoutePath, which configures the Navigator with the corresponding Pages.   |
+| iOS (launched)           | pushRoute is called                                                 | Path is parsed, and the Navigator is configured with a new set of Pages.                                                                                                                                   |
+| Android (launched)       | pushRoute is called                                                 | Path is parsed, and the Navigator is configured with a new set of Pages.                                                                                                                                   |
```

Comment:
This table currently doesn't look very good, maybe there's a better way to format this?